funderfinder/get_funders.py

```diff
@@ -0,0 +1,48 @@
+import argparse
+from datetime import datetime
+
+from sources.config import PRODUCTION_FINDERS
+
+"""
+Retrieves all funding information for a project from supported sources
+"""
+
+
+def get_project_funders(repo_name: str, project_name: str) -> list:
+    """
+    Attempts to retrieve funding data from each source for matching projects. When funding sources are found, adds the
+    source's name, a boolean is_funded field with value True, and the date the funding data was retrieved to the
+    metadata of each source of funding that was found
+    :param repo_name: Github identifier for the project (e.g. georgetown-cset/funder-finder)
+    :param project_name: Name of the project (e.g. Funder Finder)
+    :return: An array of funding metadata
+    """
+    project_funders = []
+    for finder_class in PRODUCTION_FINDERS:
+        finder = finder_class()
+        funding = finder.run(repo_name, project_name)
+        if funding:
+            funding["type"] = finder_class.name
+            funding["is_funded"] = True
+            funding["date_of_data_collection"] = datetime.now().strftime("%Y-%m-%d")
+            project_funders.append(funding)
+    return project_funders
+
+
+if __name__ == "__main__":
+    parser = argparse.ArgumentParser()
+    parser.add_argument(
+        "--repo_name",
+        help="Identifier for GitHub repo, in the form `owner_name/repo_name` "
+        "(e.g. georgetown-cset/funder-finder)",
+    )
+    parser.add_argument(
```

Comment:
I don't understand this argument. Why is the GitHub repo alone insufficient? I'm sure there is a good reason, but if I am confused, I bet a user that is not a developer will be confused too.

Reply:
Well, what I had in mind at the time was that in some cases we would be able to look a project up by name as well as by repo slug. Now that I'm coming back to this, though, I don't actually think it's a good idea to allow both options in this script, at least not right now - I can definitely see how this would be confusing, and it increases the complexity of the project without much benefit. I think I'm going to just get rid of this arg for now. Good comment!

Reply:
Resolved via 10f5442fbb714954d48a0f24cc69f4ccd1693a03